src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -295,10 +295,13 @@ def instance(userid: Optional[str] = None,
 
                     # Iterate through _daemons.values() to find the daemon with the
                     # matching zonegroup_id
-                    for daemon in RgwClient._daemons.values():
-                        if daemon.zonegroup_id == default_zonegroup:
-                            daemon_name = daemon.name
-                            break
+                    if default_zonegroup:
+                        for daemon in RgwClient._daemons.values():
+                            if daemon.zonegroup_id == default_zonegroup:
+                                daemon_name = daemon.name
+                                break
+                    else:
```

Comment:
`daemon_name = next(iter(daemon_keys))` is being repeated three times here. Not ideal

Reply:
Updated